### ISSUE TITLE
Process.Start fails to execute properly

### ISSUE THREAD

**Issue by jkotas** [dotnet/corefx#27165]
_From @victorsample on February 15, 2018 14:44_

Using Visual Studio 2017 Community
Process.Start (<name>,<command_line)
fails to execute properly.
Trying to start Postgresql utility "psql" with command line "-U aobadmin aobdb <C:\\Temp\\backupdb.sql" does not execute properly.
Putting the execution in a ".bat" file and building the command line in the bat file does work.
Apparently the form of the "psql" command is causing a problem.

_Copied from original issue: dotnet/coreclr#16399_

**Comment by danmosemsft** [comment]
@victorsample could you please share a minimal repro case, using some common tool ? 

I did this
`Process.Start(@"less.exe", @"< file.txt");`
and less opened showing the content of file.txt.

(I actually didn't realize that redirection worked here -- possibly it's less doing it?)

**Comment by wfurt** [comment]
at least on Unix, the redirection works only when executed with shell. 
Less will also show treat arguments as files to read.  It also depends how we handle escaping.

**Comment by victorsample** [comment]
A simple c# program that reads from stdin via Console.ReadLine () will demonstrate the issue.
When run from a command line or started from within Visual Studio the redirection to StdIn works properly.
When started within a dotnet assembly via Process.Start the redirection is not done.
The redirection symbol "<" and the redirection file are treated as standard arguments passed to the started program in args ().

**Comment by victorsample** [comment]
If I replace the started program with a .bat file and pass in the parms to the batch file
I can construct the original command line with the redirection and execute the actual target program. 
That does work.
Its a simple but not very elegant work around to the Process.Start deficiency.

**Comment by dmtscv** [comment]
Hi, I will try to tackle this one.

**Comment by karelz** [comment]
Yeah, this looks a bit fishy. We should not parse the args IMO. We already have APIs to redirect standard input/output/error of a new process.

I think the only remaining interesting thing here is to find out why simple code works:
```c#
Process.Start(@"less.exe", @"< file.txt");
```
And more complex does not (or find out when it does not work - e.g. args being `myarg < file.txt`).

BTW: @wtgodbe what did you have in mind for this issue? Can you please share your thoughts on next steps?

**Comment by dmtscv** [comment]
Less is working because it expects list of files as input, so the format actually is `less.exe file1.txt file2.txt [fileN.txt]`. So symbol "<" is treated as file name and after escaping it is empty, therefore ignored. To prove that, try executing `less.exe ^ file.txt` or `less.exe " test.txt` you will get same result (version 530, taken from [here](http://www.greenwoodsoftware.com/less/download.html) ). But if you execute `less , test.txt` comma won't be escaped and you will see error message ",: No such file or directory", press Enter and second file will be opened.

Now if you really want to redirect input cmd-style you can do it in following way (pay attention to quotes around command in second parameter):

`Process.Start(@"C:\Windows\System32\cmd.exe", @"/k ""PathTo\YourApp.exe < PathTo\in.txt""");`

But then you cannot redirect output using Process class. So better to stick to the means which Process class provides you with.
